Source/Core/Common/Config/Layer.h

```diff
@@ -116,7 +122,7 @@ class Layer
   }
 
   template <typename T>
-  void Set(const ConfigInfo<T>& config_info, const T& value)
+  void Set(const ConfigInfo<T>& config_info, const typename detail::Identity<T>::type& value)
```

Comment:
Never seen this before, so I googled it. And [cppreference seems to have an alternative](http://en.cppreference.com/w/cpp/language/template_argument_deduction#Non-deduced_contexts): `decltype(*std::declval<T>())`
Not sure if that works here, but it's equally (un)readable yet needs no extra struct.

Thoughts?

Reply:
It didn't seem to work, but I stumbled upon [std::common_type](https://en.cppreference.com/w/cpp/types/common_type) which seems to do what we want, so I've replaced `detail::Identity` with `std::common_type_t`.